docroot/modules/custom/foia_migrate/src/data-groomer.php

```diff
@@ -300,3 +301,113 @@ function remove_departments_property_from_agencies(array &$agency_data) {
+      if ($latest_data) {
+        $component->latest_request_time_stats = $latest_data['data'];
+        $component->latest_request_time_stats_year = $latest_data['year'];
+      }
+    }
+
+    // Add empty values for any statistics missing data.
+    $latest_request_time_stats_array = (array) $component->latest_request_time_stats;
+    $component->latest_request_time_stats = (object) insert_empty_days_values($latest_request_time_stats_array);
+
+    if (!isset($component->latest_request_time_stats_year)) {
+      $component->latest_request_time_stats_year = '';
+    }
+
+  }
+}
+
+/**
+ * Extract that last item in the Request Times Stats data object from an Agency.
+ *
+ * @param string $agency_name
+ *   String representing the name of the Agency.
+ * @param object $agency_data
+ *   An object containing the Agency information.
+ */
+function extract_processing_data_from_agency($agency_name, $agency_data) {
+  foreach ($agency_data as $agency) {
+    if ($agency->name === $agency_name && isset($agency->request_time_stats)) {
+      $latest_data = end($agency->request_time_stats);
+      $year = key($agency->request_time_stats);
+      return ['data' => $latest_data, 'year' => $year];
+    }
+  }
+}
+
+/**
+ * Adds empty values for any statistics missing data.
+ *
+ * @param array $latest_request_time_stats_array
+ *   An array containing values for component statistics.
+ */
+function insert_empty_days_values(array $latest_request_time_stats_array) {
+  $fields = [
+    'complex_average_days',
+    'complex_highest_days',
+    'complex_lowest_days',
+    'complex_median_days',
+    'expedited_average_days',
+    'expedited_highest_days',
+    'expedited_lowest_days',
+    'expedited_median_days',
+    'simple_average_days',
+    'simple_highest_days',
+    'simple_lowest_days',
+    'simple_median_days',
+  ];
+  foreach ($fields as $field) {
+    if (!array_key_exists($field, $latest_request_time_stats_array)) {
+      $latest_request_time_stats_array[$field] = '';
+    }
+  }
+  // Cleans up any array items with key of '0'.
+  if (array_key_exists('0', $latest_request_time_stats_array)) {
+    unset($latest_request_time_stats_array['0']);
+  }
+  $latest_request_time_stats_array = remove_processing_from_expedited_fields($latest_request_time_stats_array);
+  return $latest_request_time_stats_array;
+}
+
+/**
+ * Removes processing from expedited field names.
+ *
+ * @param array $latest_request_time_stats_array
+ *   An array of most recent year statistics.
+ */
+function remove_processing_from_expedited_fields(array $latest_request_time_stats_array) {
+  $keys = array_keys($latest_request_time_stats_array);
+  // Transform keys that begin w/'expedited_processing' to begin w/'expedited'.
+  if (in_array('expedited_processing_average_days', $keys) ||
+    in_array('expedited_processing_highest_days', $keys) ||
+    in_array('expedited_processing_lowest_days', $keys) ||
+    in_array('expedited_processing_median_days', $keys)
```

Comment:
I don't think we need to do this additional processing. We should probably just update the `processing_data.yml` file to look for the fields with `processing` in the name, rather than changing the name to match what we have in the yml file. The less we do to the original data the better.
e.g:
```
    -
      name: expedited_average_days
      label: 'Expedited Average Days'
      selector: /latest_request_time_stats/expedited_average_days
```